### ISSUE TITLE
compile break

### ISSUE THREAD

**Issue by Yang507** [SHUCV/digit#7]
Hello, I put your data, examples and model in official caffe-master directory.After I make all and make pycaffe, I use command "python solve.py", there is some mistake:
  
    I0818 17:01:36.724838  8188 net.cpp:106] Creating Layer score4
    I0818 17:01:36.724850  8188 net.cpp:454] score4 <- pool3
    I0818 17:01:36.724866  8188 net.cpp:411] score4 -> score4
    I0818 17:01:36.725581  8188 net.cpp:150] Setting up score4
    I0818 17:01:36.725605  8188 net.cpp:157] Top shape: 1 11 12 26 (3432)
    I0818 17:01:36.725615  8188 net.cpp:165] Memory required for data: 40601504
    I0818 17:01:36.725631  8188 layer_factory.hpp:77] Creating layer upscore_new
    I0818 17:01:36.725651  8188 net.cpp:106] Creating Layer upscore_new
    I0818 17:01:36.725662  8188 net.cpp:454] upscore_new <- score4
    I0818 17:01:36.725679  8188 net.cpp:411] upscore_new -> upscore_new
    I0818 17:01:36.725915  8188 net.cpp:150] Setting up upscore_new
    I0818 17:01:36.725937  8188 net.cpp:157] Top shape: 1 11 104 216 (247104)
    I0818 17:01:36.725949  8188 net.cpp:165] Memory required for data: 41589920
    I0818 17:01:36.725968  8188 layer_factory.hpp:77] Creating layer score
    F0818 17:01:36.725996  8188 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) 

     Unknown layer type: Crop (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm,
     BatchReindex,   Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Dropout, DummyData, 
     Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, 
     Im2col,ImageData, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, 
     MultinomialLogisticLoss, PReLU, Pooling, Power, ReLU, Reduction, Reshape, SPP, Sigmoid, 
     SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, 
     WindowData)
     *** Check failure stack trace: ***

**Comment by Yang507** [comment]
If I use your code,I cannot compile pycaffe and cannot find the module "_caffe"

**Comment by jy9387** [comment]
Our code is self-contained, which means you should use our model as well as our released code instead of the official branch of caffe.
